src/libraries/System.Private.CoreLib/src/System/SpanHelpers.cs

```diff
@@ -462,8 +484,17 @@ public static void Reverse(ref int buf, nuint length)
                     //     +---------------+
                     //     | D | C | B | A |
                     //     +---------------+
-                    tempFirst = Sse2.Shuffle(tempFirst, 0b00_01_10_11);
-                    tempLast = Sse2.Shuffle(tempLast, 0b00_01_10_11);
+                    if (Sse2.IsSupported)
+                    {
+                        // Prefer shuffling doubleword as it is faster than shuffling bytes on older systems
+                        tempFirst = Sse2.Shuffle(tempFirst, 0b00_01_10_11);
+                        tempLast = Sse2.Shuffle(tempLast, 0b00_01_10_11);
+                    }
+                    else
+                    {
+                        tempFirst = SimdShuffle(tempFirst.AsByte(), reverseMask).AsInt32();
```

Comment:
Actually we already have Vector128.Shuffle that should work fine with constant vectors representing indices (but make sure it's constant vector is passed directly, not by a local)

Reply:
Done